main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-blockservice"
+	offline "github.com/ipfs/go-ipfs-exchange-offline"
+	"github.com/ipfs/go-libipfs/gateway"
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.Flags().StringP("orchestrator", "o", "", "Saturn nodes orchestrator")
+	rootCmd.Flags().StringP("logging-endpoint", "l", "", "Saturn logging endpoint")
+	rootCmd.Flags().StringSliceP("kubo-rpc", "k", []string{}, "Kubo RPC nodes that will handle /api/v0 requests (can be set multiple times)")
+	rootCmd.Flags().IntP("port", "p", 8080, "Port to listen to")
+
+	rootCmd.MarkFlagRequired("orchestrator")
+	rootCmd.MarkFlagRequired("logging-endpoint")
+	rootCmd.MarkFlagRequired("kubo-rpc")
+}
+
+var rootCmd = &cobra.Command{
+	Use:               "bifrost",
+	CompletionOptions: cobra.CompletionOptions{DisableDefaultCmd: true},
+	Short:             "Eagle is a website CMS",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		orchestrator, err := cmd.Flags().GetString("orchestrator")
+		if err != nil {
+			return err
+		}
+
+		loggingEndpoint, err := cmd.Flags().GetString("logging-endpoint")
+		if err != nil {
+			return err
+		}
+
+		kuboRPC, err := cmd.Flags().GetStringSlice("kubo-rpc")
+		if err != nil {
+			return err
+		}
+
+		port, err := cmd.Flags().GetInt("port")
+		if err != nil {
+			return err
+		}
+
+		blockStore, err := newBlockStore(orchestrator, loggingEndpoint)
+		if err != nil {
+			return err
+		}
+
+		// Sets up an offline (no exchange) blockService based on the Saturn block store.
+		blockService := blockservice.New(blockStore, offline.Exchange(blockStore))
+
+		// // Sets up the routing system, which will proxy the IPNS routing requests to the given gateway.
+		// routing := newProxyRouting(*gatewayUrlPtr, nil)
```

Comment:
@hacdias I've filled https://github.com/protocol/bifrost-infra/issues/2327 to expose `/api/v0/routing/get`  for now.